src/block_cache/block_io.c

```diff
@@ -260,6 +260,102 @@ __wt_blkcache_read(WT_SESSION_IMPL *session, WT_ITEM *buf, WT_PAGE_BLOCK_META *b
     return (ret);
 }
 
+int
+__wt_blkcache_read_multi(WT_SESSION_IMPL *session, WT_ITEM **buf, size_t *buf_count,
+  WT_PAGE_BLOCK_META *block_meta, const uint8_t *addr, size_t addr_size)
+{
+    WT_BLOCK_DISAGG_HEADER *blk;
+    WT_BM *bm;
+    WT_BTREE *btree;
+    WT_DECL_RET;
+    WT_ITEM results[WT_DELTA_LIMIT];
+    WT_ITEM *tmp;
+    WT_PAGE_BLOCK_META block_meta_tmp;
+    const WT_PAGE_HEADER *dsk;
+    uint32_t count, i;
+
+    btree = S2BT(session);
+    bm = btree->bm;
+
+    /* Skip block cache for M2, just read the base + delta pack. */
+    count = WT_ELEMENTS(results);
+
+    if (bm->read_multiple == NULL) {
+        WT_RET(__wt_calloc_def(session, 1, &tmp));
+        WT_CLEAR(tmp[0]);
+        WT_ERR(__wt_blkcache_read(session, &tmp[0], block_meta, addr, addr_size));
+        *buf_count = 1;
+        *buf = tmp;
+        return (0);
+    }
+
+    WT_CLEAR(results);
+    WT_RET(bm->read_multiple(bm, session, &block_meta_tmp, addr, addr_size, &results[0], &count));
+    WT_ASSERT(session, count > 0);
+
+    /*
+     * For the base image, we have a structure like this:
+     *
+     * ------------------------
+     * | page header          |
+     * ------------------------
+     * | block header         |
+     * ------------------------
+     * | data                 |
+     * ------------------------
+     *
+     * In this case, the encryption/compression flags live in the page header.
+     */
+    dsk = results[0].data;
+    if (F_ISSET(dsk, WT_PAGE_ENCRYPTED)) {
+        WT_ASSERT(session, session == NULL);
```

Comment:
I don't understand this...?   If the page is encrypted, then the session must be NULL?  But the session can't be null, it's been used earlier in the function??  I'm missing something basic.

Reply:
This one is my fault sorry! It's basically a `WT_ASSERT(session, false)` since encryption and compression are still WIP in another branch. I use that weird form since it avoids spurious "unreachable code" warnings in newer compilers.